grpc/tests/src/test/system_costs.rs

```diff
@@ -0,0 +1,927 @@
+        UpgradeRequestBuilder, DEFAULT_ACCOUNTS, DEFAULT_ACCOUNT_PUBLIC_KEY, DEFAULT_PAYMENT,
+        DEFAULT_PROTOCOL_VERSION, DEFAULT_RUN_GENESIS_REQUEST,
+    },
+    AccountHash, DEFAULT_ACCOUNT_ADDR, DEFAULT_ACCOUNT_INITIAL_BALANCE,
+    MINIMUM_ACCOUNT_CREATION_BALANCE,
+};
+use casper_execution_engine::{
+    core::engine_state::{upgrade::ActivationPoint, GenesisAccount, SYSTEM_ACCOUNT_ADDR},
+    shared::{
+        host_function_costs::{
+            HostFunction, HostFunctionCosts, DEFAULT_TRANSFER_FROM_PURSE_TO_PURSE_COST,
+        },
+        motes::Motes,
+        opcode_costs::OpcodeCosts,
+        storage_costs::StorageCosts,
+        system_config::{
+            auction_costs::{
+                AuctionCosts, DEFAULT_ADD_BID_COST, DEFAULT_DELEGATE_COST, DEFAULT_DISTRIBUTE_COST,
+                DEFAULT_RUN_AUCTION_COST, DEFAULT_SLASH_COST, DEFAULT_UNDELEGATE_COST,
+                DEFAULT_WITHDRAW_BID_COST, DEFAULT_WITHDRAW_DELEGATOR_REWARD_COST,
+                DEFAULT_WITHDRAW_VALIDATOR_REWARD_COST,
+            },
+            mint_costs::{
+                MintCosts, DEFAULT_BALANCE_COST, DEFAULT_MINT_COST,
+                DEFAULT_REDUCE_TOTAL_SUPPLY_COST, DEFAULT_TRANSFER_COST,
+            },
+            proof_of_stake_costs::{
+                ProofOfStakeCosts, DEFAULT_FINALIZE_PAYMENT_COST, DEFAULT_SET_REFUND_PURSE_COST,
+            },
+            standard_payment_costs::{StandardPaymentCosts, DEFAULT_PAY_COST},
+            SystemConfig,
+        },
+        wasm,
+        wasm_config::{WasmConfig, DEFAULT_MAX_STACK_HEIGHT, DEFAULT_WASM_MAX_MEMORY},
+    },
+    storage::protocol_data::DEFAULT_WASMLESS_TRANSFER_COST,
+};
+use casper_types::{
+    auction::{self, DelegationRate},
+    mint, proof_of_stake, runtime_args,
+    system_contract_type::AUCTION,
+    ProtocolVersion, PublicKey, RuntimeArgs, SecretKey, U512,
+};
+
+const SYSTEM_CONTRACT_HASHES_NAME: &str = "system_contract_hashes.wasm";
+const CONTRACT_TRANSFER_TO_ACCOUNT: &str = "transfer_to_account_u512.wasm";
+const CONTRACT_ADD_BID: &str = "add_bid.wasm";
+
+const VALIDATOR_1_SECRET_KEY: Lazy<SecretKey> =
+    Lazy::new(|| SecretKey::ed25519([123; SecretKey::ED25519_LENGTH]));
+const VALIDATOR_1: Lazy<PublicKey> = Lazy::new(|| PublicKey::from(&*VALIDATOR_1_SECRET_KEY));
+static VALIDATOR_1_ADDR: Lazy<AccountHash> = Lazy::new(|| AccountHash::from(&*VALIDATOR_1));
+const VALIDATOR_1_STAKE: u64 = 250_000;
+const BOND_AMOUNT: u64 = 42;
+const BID_AMOUNT: u64 = 99;
+const TRANSFER_AMOUNT: u64 = 123;
+const BID_DELEGATION_RATE: DelegationRate = 123;
+
+const NEW_ADD_BID_COST: u32 = DEFAULT_ADD_BID_COST * 2;
+const NEW_WITHDRAW_BID_COST: u32 = DEFAULT_WITHDRAW_BID_COST * 3;
+const NEW_DELEGATE_COST: u32 = DEFAULT_DELEGATE_COST * 4;
+const NEW_UNDELEGATE_COST: u32 = DEFAULT_UNDELEGATE_COST * 5;
+const DEFAULT_ACTIVATION_POINT: ActivationPoint = 1;
+
+static OLD_PROTOCOL_VERSION: Lazy<ProtocolVersion> = Lazy::new(|| *DEFAULT_PROTOCOL_VERSION);
```

Comment:
```suggestion
static OLD_PROTOCOL_VERSION: Lazy<ProtocolVersion> = DEFAULT_PROTOCOL_VERSION;
```

Reply:
Can't do. Unfortunately Lazy object is not copyable

Reply:
Gotcha